vignettes/convert.qmd

```diff
@@ -15,13 +15,70 @@ execute:
+
+## How to choose between DuckDB, Parquet, and CSV {#duckdb-vs-parquet-csv}
+
+### Speed {#speed-comparison}
+
+In our testing (see @fig-csv-duckdb-parquet-speed), using `DuckDB` database has a significant speed advantage over running the analysis on raw CSV.gz files. Using the full origin destination dataset for 18 months, we compared a query that would find mean hourly number of trips between all zone pairs over the whole time period. `DuckDB` database was up to 8 times faster than running the analysis on raw CSV.gz files, and up to 5 times faster than running the analysis on `Parquet` files.
+
+![Data processing speed comparison: DuckDB engine running on CSV.gz files vs DuckDB database vs a folder of Parquet files](media/duckdb-parquet-csv-speed-mean-hourly-v1.svg){#fig-csv-duckdb-parquet-speed}
+
+
+<details><summary>You can see the query we used for measuring the speed here</summary>
+
+For reference, here is a simple query we used for speed comparison in @fig-csv-duckdb-parquet-speed:
+
+```r
+# data represents either CSV files acquired from `spod_get()`, a `DuckDB` database or a folder of Parquet files connceted with `spod_connect()`
+data |>
+  group_by(id_origin, id_destination, time_slot) |> 
+  summarise(mean_hourly_trips = mean(n_trips, na.rm = TRUE),
+    .groups = "drop")
+```
+
+</details>
+
+
+@fig-csv-duckdb-parquet-speed also shows that `DuckDB` format will give you the best performance even on low-end systems with limited memory and number of processor cores, conditional on a fast SSD storage. Also note, that if you do choose to work with long time periods using CSV.gz files via `spod_get()`, you will need to balance the amount of memory and processor cores via the `max_n_cpu` and `max_mem_gb` arguments, otherwise the analysis may fail (see the grey area in the figure), when there are too many parallel processes running at the same time with limited memory.
+
+### Convenience
+
+The approaches to working with all formats are exactly the same, as under the hood all analysis is actually done by the `DuckDB` [@duckdb-r] engine. So from that point of view, there is no difference between data formats. You can manipulate the data using `{dplyr}` functions such as `select()`, `filter()`, `mutate()`, `group_by()`, `summarise()`, etc. In the end of any sequence of commands you will need to add `collect()` to execute the whole chain of data manipulations and load the results into memory in an R `data.frame`/`tibble`. We provide examples in the following sections.
+
+### Use cases
```

Comment:
by use cases I meant what is the approach to working with this data. Iterative (download data, analyse, download more data, analyse), or bulk (download all). I would not say this is dependent on the time period, even though the examples i give below are connected with time.

Reply:
alternative to insert before the bullet points 
"### Scenarios of getting the data

The choice between converting to `DuckDB` and `Parquet` could also be made based on how you plan to work with the data. Specifically whether you want to just download long periods or even all available data, or if you want to get the data gradually, as you progress through with the analysis."

Reply:
OK, I understand. But still, the different approach (iteration/bulk) depends on the weight of the data, right? I just found that the title 'case uses' is meaningless (or at least unclear) because each analysis is always a different case use, so which here the aspect that makes you choose one or another option?

Reply:
Sorry, we crossed messages. My previous message was answering >by use cases I meant what is the approach to working with this data. Iterative (download data, analyse, download more data, analyse), or bulk (download all). I would not say this is dependent on the time period, even though the examples i give below are connected with time.

Reply:
This new text before the bullet points sounds perfect!

Reply:
@eugenividal good

Reply:
"Data collection scenarios" as a title not a big fan, but I can't think of anything better.